elements/lisk-chain/src/validate.ts

```diff
@@ -186,10 +186,20 @@ export const validateGenesisBlockHeader = (block: GenesisBlock, accountSchema: S
 	}
 
 	const accountAddresses = header.asset.accounts.map(a => a.address);
-
-	if (!objects.bufferArrayOrderByLex(accountAddresses)) {
+	const copiedAddresses = [...accountAddresses];
+	copiedAddresses.sort((a, b) => {
+		if (a.length > b.length) {
+			return 1;
+		}
+		if (a.length < b.length) {
+			return -1;
+		}
+		return a.compare(b);
+	});
+
+	if (!objects.bufferArrayEqual(accountAddresses, copiedAddresses)) {
 		errors.push({
-			message: 'should be lexicographically ordered',
+			message: 'should be length and lexicographically ordered',
```

Comment:
```suggestion
			message: 'should be ordered lexicographically and by length',
```

Reply:
I put length first because length takes priority

Reply:
Then it should be 

> should be ordered by length and lexicographically